--- /Users/mahimachaudhary/Downloads/eval4_linux/drivers/input/input.c	2026-07-15 14:00:18
+++ /Users/mahimachaudhary/Downloads/eval4_linux_6_6_25/drivers/input/input.c	2026-07-15 15:15:56
@@ -85,7 +85,7 @@
 	return value;
 }
 
-static int input_clamp_abs_value(struct input_dev *dev,
+static noinline int input_clamp_abs_value(struct input_dev *dev,
 					 unsigned int code, int *value)
 {
 	volatile struct input_absinfo absnew;
@@ -233,7 +233,7 @@
 #define INPUT_FLUSH		8
 #define INPUT_PASS_TO_ALL	(INPUT_PASS_TO_HANDLERS | INPUT_PASS_TO_DEVICE)
 
-static int input_handle_abs_event(struct input_dev *dev,
+static noinline int input_handle_abs_event(struct input_dev *dev,
 				  unsigned int code, int *pval)
 {
 	struct input_mt *mt = dev->mt;
@@ -762,7 +762,7 @@
 
 	__input_release_device(handle);
 
-	if (!dev->inhibited && !--dev->users) {
+	if (!--dev->users && !dev->inhibited) {
 		if (dev->poller)
 			input_dev_poller_stop(dev->poller);
 		if (dev->close)
